src/core/credentials/transport/tls/grpc_tls_certificate_provider.cc

```diff
@@ -99,14 +99,28 @@ absl::Status ValidatePemKeyCertPair(absl::string_view cert_chain,
   return absl::OkStatus();
 }
 
+bool HasRootCertInfoChanged(
+    const absl::StatusOr<std::shared_ptr<RootCertInfo>>& old,
+    const absl::StatusOr<std::shared_ptr<RootCertInfo>>& updated) {
+  if (old.status() != updated.status()) return true;  // Status changed.
+  if (!old.ok()) return false;  // Both have same non-OK status.
+  // Both have OK status.
+  if (*old == nullptr) {
+    return *updated != nullptr;
+  } else {
```

Comment:
No need for the `else`, since the `if` block returns.  Can just say:

```
if (*old == nullptr) return *updated != nullptr;
if (*updated == nullptr) return true;
```